napari/_qt/tree/qt_tree_model.py

```diff
@@ -0,0 +1,369 @@
+
+    def canDropMimeData(
+        self,
+        data: QMimeData,
+        action: Qt.DropAction,
+        row: int,
+        column: int,
+        parent: QModelIndex,
+    ) -> bool:
+        """Returns true if a model can accept a drop of the data.
+
+        The default implementation only checks if data has at least one format in the list
+        of mimeTypes() and if action is among the model's supportedDropActions().
+
+        Reimplementing this function lets us test whether the data can be dropped at
+        ``row``, ``column``, ``parent`` with ``action``.  Here, we just check that
+        ``parent`` is a :class:`Group`.
+
+        Returns
+        -------
+        bool
+            Whether we can accept a drop of data at ``row``, ``column``, ``parent``
+        """
+        return self.getItem(parent).is_group()
+
+    def columnCount(self, parent: QModelIndex) -> int:
+        """Returns the number of columns for the children of the given ``parent``.
+
+        In a tree, the number of columns is always 1.
+        """
+        return 1
+
+    def data(self, index: QModelIndex, role: Qt.ItemDataRole):
+        """Return data stored under ``role`` for the item at ``index``.
+
+        A given class:`QModelIndex` can store multiple types of data, each with its own
+        "ItemDataRole".
+        """
+        item = self.getItem(index)
+        if role == Qt.DisplayRole:
+            return str(item.name)
+        if role == Qt.UserRole:
+            return self.getItem(index)
+        return None
+
+    def dropMimeData(
+        self,
+        data: QMimeData,
+        action: Qt.DropAction,
+        destRow: int,
+        col: int,
+        parent: QModelIndex,
+    ) -> bool:
+        """Handles ``data`` from a drag and drop operation that ended with ``action``.
+
+        The specified row, column and parent indicate the location of an item in the model
+        where the operation ended. It is the responsibility of the model to complete the
+        action at the correct location.
+
+        Returns
+        -------
+        bool
+            ``True`` if the ``data`` and ``action`` were handled by the model;
+            otherwise returns ``False``.
+        """
+        if not data or action != Qt.MoveAction:
+            return False
+        if not data.hasFormat(self.mimeTypes()[0]):
+            return False
+
+        if isinstance(data, NodeMimeData):
+            dest_idx = self.getItem(parent).index_from_root()
+            dest_idx = dest_idx + (destRow,)
+            moving_indices = data.node_indices()
+
+            logger.debug(
+                f"dropMimeData: indices {moving_indices} ➡ {dest_idx}"
+            )
+
+            if len(moving_indices) == 1:
+                self._root.move(moving_indices[0], dest_idx)
+            else:
+                self._root.move_multiple(moving_indices, dest_idx)
+
+        # XXX: returning False seems to go against the Qt docs instructions, but
```

Comment:
Does this still need to be addressed or is this determined now?

Reply:
fixed in the next pr... just backported to this one.